### PR TITLE
test to ensure that celery does not scale down by negative values

### PR DESCRIPTION
**Pull request by komuW**
This is in relation to this issue: https://github.com/celery/celery/issues/1967

So, it looks like that issue has been fixed in the current master(tip) but there has not been any testcase. This fixes that.
@insertjokehere had intimated that he had a fix but didn't have a good testcase; https://github.com/celery/celery/issues/1967#issuecomment-196491985 

So the gist of this pull request is to add the neccesary testcase

### CODE REVIEW

**Comment by thedrow** [comment]
This needs to be rebased. I'll review it later.

**Comment by komuW** [comment]
Hi @thedrow.
I don't mean to bug you, but have you had time to review my pull request? 
I would appreciate any feedback.

**Comment by thedrow** [comment]
I didn't. I'll have time in a bit. Thanks for the ping.

File: celery/worker/autoscale.py [hunk 1]
```diff
@@ -133,11 +135,13 @@ def scale_down(self, n):
 
     def _grow(self, n):
         info('Scaling up %s processes.', n)
+        self.growth_numbers.append(n)
```
Comment (ask) [inline celery/worker/autoscale.py]: I think this part should be in the test, not in the original class

**Comment by thedrow** [comment]
LGTM. Please rebase and I'll merge.

**Comment by komuW** [comment]
@thedrow I have rebased. 
Also following a comment by @ask , I've refactored the test to be self-sufficient and thus this pull request does not make any changes to celery/worker/autoscale.py

**Pull request merged**